src/util/strutil.h

```diff
@@ -6,11 +6,11 @@
 
 #pragma once
 
-//convert str to a lower case string and save it in lower
-void str_tolower(const char *str, char *lower, size_t *lower_len);
+//convert str to a lower case string and save it in lower, pass "" for the root locale or NULL for the default locale
+void str_tolower(const char *str, char *lower, size_t *lower_len, const char *locale);
```

Comment:
Due to the overhead and the distinction between load-time and runtime invocations
I prefer if we introduce two additional functions:
```
void str_tolower(const char *str, char *lower, size_t *lower_len);
void str_tolower_locale(const char *str, char *lower, size_t *lower_len, const char *locale);
void str_toupper(const char *str, char *upper, size_t *upper_len);
void str_toupper_locale(const char *str, char *upper, size_t *upper_len, const char *locale);
```

The original `str_toupper` and `str_tolower` should be called when we explicitly pass NULL for locale in this PR.
e.g.
`str_tolower(procedure, lowercase_proc_name, &lowercase_proc_name_len, NULL);` -> `str_tolower(procedure, lowercase_proc_name, &lowercase_proc_name_len);`

Calls to `toUpper` or `toLower` made by the user at runtime will use the new `str_tolower_locale` and `str_toupper_locale` versions.